File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/converter/String2StringConverter.java
package com.cedarsoft.commons.javafx.converter;

import javafx.util.StringConverter;

/**
 * Converter that simply returns the string itself.
 * Can be used to avoid null checks with converters
 */
public class String2StringConverter extends StringConverter<String> {
  @Override
  public String toString(String object) {
    return object;
  }

  @Override
  public String fromString(String string) {
    return string;
  }
}

File: commons/javafx/src/main/kotlin/com/cedarsoft/commons/javafx/converter/String2StringConverter.kt
package com.cedarsoft.commons.javafx.converter

import javafx.util.StringConverter

/**
 * Converter that simply returns the string itself.
 * Can be used to avoid null checks with converters
 */
class String2StringConverter : StringConverter<String>() {
  override fun toString(value: String): String {
    return value
  }

  override fun fromString(string: String): String {
    return string
  }
}
